pkg/proc/target_exec.go

```diff
@@ -999,6 +1010,107 @@ func setStepIntoBreakpoint(dbp *Target, curfn *Function, text []AsmInstruction,
 	return nil
 }
 
+// setStepIntoNewProcBreakpoint sets a temporary breakpoint on
+// runtime.newproc that, when hit, clears all temporary breakpoints and sets
+// a new temporary breakpoint on the starting function for the new
+// goroutine.
+func setStepIntoNewProcBreakpoint(p *Target, sameGCond ast.Expr) {
+	const (
+		runtimeNewprocFunc1 = "runtime.newproc.func1"
+		runtimeRunqput      = "runtime.runqput"
+	)
+	rnf := p.BinInfo().LookupFunc()[runtimeNewprocFunc1]
+	if len(rnf) != 1 {
+		logflags.DebuggerLogger().Error("could not find " + runtimeNewprocFunc1)
+		return
+	}
+	text, err := Disassemble(p.Memory(), nil, p.Breakpoints(), p.BinInfo(), rnf[0].Entry, rnf[0].End)
+	if err != nil {
+		logflags.DebuggerLogger().Errorf("could not disassemble "+runtimeNewprocFunc1+": %v", err)
+		return
+	}
+
+	callfile, callline := "", 0
+	for _, instr := range text {
+		if instr.Kind == CallInstruction && instr.DestLoc != nil && instr.DestLoc.Fn != nil && instr.DestLoc.Fn.Name == runtimeRunqput {
+			callfile = instr.Loc.File
+			callline = instr.Loc.Line
+			break
+		}
+	}
+	if callfile == "" {
+		logflags.DebuggerLogger().Error("could not find " + runtimeRunqput + " call in " + runtimeNewprocFunc1)
+		return
+	}
+	var pcs []uint64
```

Comment:
We only ever use `pcs[0]` after this, so why not just a `var pc uint64` and assign the first address where `pcstmt.Stmt == true`?

Reply:
Done.